### DISCUSSION TITLE
Can't use tcp or mqtt, just ws and wss on browser

### DISCUSSION THREAD

**Discussion by jianjun-Web** [mqttjs/MQTT.js#1812] [Q&A] [answered]
when i set http or mqtt, the address of the e link is ws
![image](https://github.com/mqttjs/MQTT.js/assets/70632135/c267dab6-c344-43b2-a380-4d12c38c4a0a)
![image](https://github.com/mqttjs/MQTT.js/assets/70632135/c12618c2-8b60-436e-b57d-eb20c9c95d5d)

**Comment by jianjun-Web** [comment]
mqtts and https will instead wss，mqtt and http will instead ws

**Reply by robertsLando** [reply, marked as answer]
See: https://github.com/mqttjs/MQTT.js#browser

**Reply by jianjun-Web** [reply]
can you tell me, see which piece of content can solve this problem

**Reply by robertsLando** [reply]
If you just could spend 10 seconds to read the docs I linked above you would immediatly find out that this cannot be solved. Please click on the link above.
